### ISSUE TITLE
Chat control should support images

### ISSUE THREAD

**Issue by AlexandreSajus** [Avaiga/taipy#1314]
### Description

The chat control of #1092 is really cool! It would be a good addition to support images in the messages of the chat. This would be especially useful since now LLMs start supporting image vision and image generation. I could create cool demos around this chat control.

Florian:
People are also using chat demos to visualize tables inside their chat. This could also be a nice addition.

### Acceptance Criteria

- [ ] Ensure new code is unit tested, and check code coverage is at least 90%.
- [ ] Create related issue in taipy-doc for documentation and Release Notes.
- [ ] Check if a new demo could be provided based on this, or if legacy demos could be benefit from it.
- [ ] Ensure any change is well documented.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

**Comment by ioannastantzou** [comment]
Hello there I think I could work on this issue!

**Comment by FlorianJacta** [comment]
Great, I will assign it to you!

**Comment by ioannastantzou** [comment]
hello again! can you specify where the chat code exists?

**Comment by ioannastantzou** [comment]
> hello again! can you specify where the chat code exists?

is it taipy\frontend\taipy-gui\src\components\Taipy\chat.tsx ? @FabienLelaquais @FredLL-Avaiga

**Comment by yaten2302** [comment]
Hey @FlorianJacta @AlexandreSajus , could you please assign this issue to me, if no one is working on this?

**Comment by FlorianJacta** [comment]
@ioannastantzou is normally working on it. Could you confirm @ioannastantzou ?

**Comment by yaten2302** [comment]
> @ioannastantzou is normally working on it. Could you confirm @ioannastantzou ?

It's ok if someone is working on this, but if not I'm in 👍

**Comment by yaten2302** [comment]
Hey @FlorianJacta @AlexandreSajus , can this issue be assigned to me?

Also, I would like to ask that in here, do we've to add a files icon on the left side of the message box (similar to what is present in chatgpt)?<br>
![image](https://github.com/user-attachments/assets/a0fde0cf-65d9-496d-957d-dc7ecd906fcd)

<br>

![image](https://github.com/user-attachments/assets/8a86ec0c-e348-4a13-9001-1efcab3a4846)

**Comment by FlorianJacta** [comment]
What do you mean by a file icon?

Can this be assigned? @FredLL-Avaiga @FabienLelaquais

**Comment by jrobinAV** [comment]
We could propose an upload button near the input area that opens a file upload window. The chat control should be able to display file images. (A download button close to the image can be considered)

**Comment by mehulmathur16** [comment]
Hi @jrobinAV, can you please assign me this issue ?

**Comment by sOnU1002** [comment]
hey i can work on this please can u assign thi sto me

**Comment by FlorianJacta** [comment]
@yaten2302, I have assigned you this issue!

@sOnU1002 I have already assigned you to another issue. @mehulmathur16, there are other issues to take!

**Comment by yaten2302** [comment]
@FlorianJacta , thanks for assigning this issue 👍 
I'll create a PR for this by Tuesday or Wednesday max!

**Comment by yaten2302** [comment]
Apologies for the huge delay on this issue. Actually, I was having a doubt that in this file - `frontend/taipy-gui/src/components/Taipy/Chat.tsx`, it's showing 2 problems <br>
1. 
![image](https://github.com/user-attachments/assets/ec9511c4-f1a5-4b85-bccf-b86ad5b48f4d)

2. 
![image](https://github.com/user-attachments/assets/bca16162-1474-44cf-8729-b843d7d0d268)

I haven't changed anything in the codebase, is this issue arising because something is not installed? Like in the 1st one, it's showing that - not able to find module Grid2?

@FredLL-Avaiga @AlexandreSajus

**Comment by FredLL-Avaiga** [comment]
I think your installed version of Mui is not up-to-date. 
My advice would be to remove the `node_modules` directory and the `package-lock.json` file and reinstall everything from scratch

**Comment by yaten2302** [comment]
@FredLL-Avaiga , I re cloned the taipy repo and tried to install the dependencies, but now it's showing this error :/ <br>
![image](https://github.com/user-attachments/assets/5a8158dd-dbdc-44ff-b3cc-33692e743986)

**Comment by FredLL-Avaiga** [comment]
You first need to build taipy-gui
And then install taipy-gui in taipy with `npm i` in taipy 
Then you can build taipy

**Comment by yaten2302** [comment]
Thanks @FredLL-Avaiga , now it's working👍 

Also, I wanted to ask about [this](https://github.com/Avaiga/taipy/issues/1314#issuecomment-2294679691) comment, like for the chat to support, we need an icon to show the user to attach an image. Should I add an icon in the chat itself (as in chatGPT)?

**Comment by yaten2302** [comment]
Also, I wanted to ask that in this PR https://github.com/Avaiga/taipy/pull/1092, you took a screenshot of the chat element, could you please guide that if I want to test the element locally, how can I view them? (as you've shown in this PR)<br>
![image](https://github.com/user-attachments/assets/d12ef924-a385-43e7-8044-2503eeee75f2)

**Comment by FredLL-Avaiga** [comment]
> Also, I wanted to ask that in this PR https://github.com/Avaiga/taipy/pull/1092, you took a screenshot of the chat element, could you please guide that if I want to test the element locally, how can I view them? (as you've shown in this PR)<br>
> ![image](https://github.com/user-attachments/assets/d12ef924-a385-43e7-8044-2503eeee75f2)
> 

You create a simple python script and run it locally
You can find simple examples in doc/examples/gui

**Comment by FredLL-Avaiga** [comment]
> Thanks @FredLL-Avaiga , now it's working👍 
> 
> Also, I wanted to ask about [this](https://github.com/Avaiga/taipy/issues/1314#issuecomment-2294679691) comment, like for the chat to support, we need an icon to show the user to attach an image. Should I add an icon in the chat itself (as in chatGPT)?

I'm not sure to understand your question. 
If it is about adding an icon button to allow the user to choose an image then yes,I think it should go in the input field besides the send icon. 
But we also want to support drag and drop and copy paste...

**Comment by yaten2302** [comment]
@FredLL-Avaiga , yes, I was asking about this only. I got it now 👍 I'll add an icon in the txt bar for adding images.
First, I'll create a draft PR for this and then if it looks good, then I'll also add support for drag and drop and copy paste.

**Comment by yaten2302** [comment]
@FredLL-Avaiga , I've a doubt, that when I'm running this command - `py -m pip install taipy-location`, it's showing this error:
```
ERROR: Could not install packages due to an OSError: [WinError 2] The system cannot find the file specified: 'C:\\Python312\\Scripts\\cftp.exe' -> 'C:\\Python312\\Scripts\\cftp.exe.deleteme
```
I tried searching about it on stackoverflow, found this solution but, still it's not wokring :/ - `py -m pip install --user taipy-location`. 
Actually, my application is still using taipy main lib even if I've changed certain things in the taipy elements(locally).

Any suggestions regarding how can I fix this?

**Comment by FredLL-Avaiga** [comment]
What is taipy-location ?

**Comment by yaten2302** [comment]
@FredLL-Avaiga ,the taipy location is: `d:/taipy` and the `taipy-prac`(in which the main.py file is present) is also in the d:/

**Comment by yaten2302** [comment]
@FredLL-Avaiga , I was trying to install my local taipy package in a sample project, but when I'm click on the the location of `import taipy.gui.builder as tgb`, it's showing this: <br>
![image](https://github.com/user-attachments/assets/9bd60861-e9af-4639-8812-82c652414dae)

But, since, my taipy file is located in `d:/taipy`, shouldn't it show this location? I tried to uninstall taipy as well and then run thes commands - `py -m pip install taipy-location` and `py -m pip install --user taipy-location`, since the 1st one was showing this error: 

```
ERROR: Could not install packages due to an OSError: [WinError 2] The system cannot find the file specified: 'C:\\Python312\\Scripts\\cftp.exe' -> 'C:\\Python312\\Scripts\\cftp.exe.deleteme
```

**Comment by Satoshi-Sh** [comment]
If you don't mind, can I start working on this issue?

**Comment by yaten2302** [comment]
@Satoshi-Sh , thanks for the offer, but could you give me some time, currently working on to fix it :)

**Comment by Satoshi-Sh** [comment]
Hi @yaten2302 , I think several people can work on the same issue. I will wait for a response from the maintainers.

**Comment by FlorianJacta** [comment]
Yes, multiple people can work on the same issue and you can also collaborate together on it

**Comment by Satoshi-Sh** [comment]
@quest-bot embark

**Comment by Satoshi-Sh** [comment]
Could this be closed? Implemented on #2078

**Comment by FredLL-Avaiga** [comment]
PR #2078
